src/scout_apm/core/socket.py

```diff
@@ -145,16 +146,21 @@ def _send(self, command, do_async=True):
 
         try:
             self.socket.sendall(self._message_length(data))
+        except (OSError) as e:
+            logger.debug("CoreAgentSocket exception on length _send: %s on PID: %s on thread: %s" % (repr(e), os.getpid(), threading.current_thread()))
+            return None
+
+        try:
             self.socket.sendall(data.encode())
         except (OSError) as e:
-            logger.debug("CoreAgentSocket exception on _send: %s" % repr(e))
+            logger.debug("CoreAgentSocket exception on data _send: %s on PID: %s on thread: %s" % (repr(e), os.getpid(), threading.current_thread()))
             return None
 
         if do_async is True:
             return True
         else:
             # TODO read respnse back in to command
-            self._read_response()
+            response = self._read_response()
```

Comment:
Save an allocation by removing `response =`

Reply:
You're right it's an unused variable, but I'm going to leave it for clarity when we start needing the responses from the core agent. It doesn't hurt anything right here.